src/plots/plots.js

```diff
@@ -752,6 +780,85 @@ plots.supplyLayoutGlobalDefaults = function(layoutIn, layoutOut) {
     coerce('smith');
 };
 
+plots.plotAutoSize = function plotAutoSize(gd, layout, fullLayout) {
+    var context = gd._context || {},
+        frameMargins = context.frameMargins,
+        newWidth,
+        newHeight;
+
+    if(typeof gd.emit === 'function') gd.emit('plotly_autosize');
```

Comment:
Good call. `Plots.supplyDefaults` should work with plain objects.

Could we make this check more robust? I'd vote for using [`Lib.isPlotDiv`](https://github.com/plotly/plotly.js/blob/19a37357341e7e098ce2c20ee98f65965c2c5d66/src/lib/index.js#L380).